client/state/analytics/middleware.js

```diff
@@ -32,13 +32,18 @@ const eventServices = {
 	adwords: ( { properties } ) => trackCustomAdWordsRemarketingEvent( properties ),
 };
 
-// Whitelists specific parameters to avoid polluting page view events
-const PAGE_VIEW_SERVICES_ALLOWED_PARAMS = [ 'client_id' ];
+// list of unsafe params that need to be blocked from beign passed down to the recorder.
+const PAGE_VIEW_SERVICES_BLOCKED_PARAMS = [ 'service' ];
+
+const omitUnsafeParams = flowRight(
```

Comment:
Unit tests would be much appreciated here. Should we also warn/error for any camel case properties?

Reply:
It appears that service is only used for the switch below, we can remove that value earlier. (Though we should audit this with each function in the switch.)
```jsx
const { service = 'default', params } = payload;

switch ( type ) {
 			case ANALYTICS_EVENT_RECORD:
 				return invoke( eventServices, service, payload );	 				return invoke( eventServices, service, payload );
//...
```

Reply:
Updated on 658415952601d26c02d344a04ee684cc08ce6548.

Adding unit tests.